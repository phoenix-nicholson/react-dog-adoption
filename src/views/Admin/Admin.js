import React from 'react';

import { useState } from 'react';
import DogForm from '../../components/DogForm';
import { createDog } from '../../services/dog';

export default function Admin() {
  const [dog, setDog] = useState({});

  const updateDogState = (key, value) => {
    dog[key] = value;
    setDog({ ...dog });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    await createDog(dog);
    if (dog !== '') {
      alert("You've successfully added a dog");
    } else {
      alert('Request to database has failed, please try again');
    }
  };

  return (
    <div>
      <DogForm {...dog} updateDogState={updateDogState} handleSubmit={handleSubmit} />
    </div>
  );
}
