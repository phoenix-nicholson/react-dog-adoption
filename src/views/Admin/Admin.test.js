import { render } from '@testing-library/react';
import Admin from './Admin';

test('renders Admin page', () => {
  const container = render(<Admin />);
  expect(container).toMatchSnapshot();
});
