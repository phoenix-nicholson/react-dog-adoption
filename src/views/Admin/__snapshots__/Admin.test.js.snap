// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Admin page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form>
          <label>
            Name:
            <input
              name="name"
              type="text"
              value=""
            />
          </label>
          <label>
            Age:
            <input
              name="age"
              type="text"
              value=""
            />
          </label>
          <label>
            Image URL:
            <input
              name="image"
              type="text"
              value=""
            />
          </label>
          <label>
            Breed:
            <input
              name="breed"
              type="text"
              value=""
            />
          </label>
          <label>
            Bio:
            <input
              name="bio"
              type="text"
              value=""
            />
          </label>
          <button>
            submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form>
        <label>
          Name:
          <input
            name="name"
            type="text"
            value=""
          />
        </label>
        <label>
          Age:
          <input
            name="age"
            type="text"
            value=""
          />
        </label>
        <label>
          Image URL:
          <input
            name="image"
            type="text"
            value=""
          />
        </label>
        <label>
          Breed:
          <input
            name="breed"
            type="text"
            value=""
          />
        </label>
        <label>
          Bio:
          <input
            name="bio"
            type="text"
            value=""
          />
        </label>
        <button>
          submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
